pilot/pkg/model/destination_rule.go

```diff
@@ -87,3 +87,106 @@ func (ps *PushContext) mergeDestinationRule(p *processedDestRules, destRuleConfi
 	p.destRule[resolvedHost] = &destRuleConfig
 	p.exportTo[resolvedHost] = exportToMap
 }
+
+// inheritDestinationRule child config inherits settings from parent mesh/namespace
+func (ps *PushContext) inheritDestinationRule(parent, child *config.Config) *config.Config {
+	if parent == nil {
+		return child
+	}
+	if child == nil {
+		return parent
+	}
+
+	parentDR := parent.Spec.(*networking.DestinationRule)
+	if parentDR.TrafficPolicy == nil {
+		return child
+	}
+	copied := child.DeepCopy()
+	mergedRule := copied.Spec.(*networking.DestinationRule)
+
+	if mergedRule.TrafficPolicy == nil {
```

Comment:
Could all this custom merging be achieved by `proto.Merge()`? if not, can you add comments about the merging semantics and how it differs?

it may end up that we do want similar logic for performance, although I doubt this would be a bottleneck, but if that is the case we should comment+test that it is actually the same behavior as proto.Merge (if that is what is desired)

Reply:
ping

Reply:
done